secureli/repositories/repo_files.py

```diff
@@ -117,3 +115,34 @@ def load_file(self, file_path: Path) -> str:
             pass
 
         raise ValueError(f"An unknown error occurred loading the file from {file_path}")
+
+    def list_staged_files(self, folder_path: Path) -> list[Path]:
+        """
+        Lists the file paths of all staged files in a repository, or raises ValueError if the provided path
+        is not a git repo
+        :param folder_path: The path to a git repo containing files
+        :raises ValueError: The specified path does not exist or is not a git repo
+        :return: The list of staged file names
+        """
+        self._confirm_is_git_repo(folder_path)
```

Comment:
I wrote similar logic for #436. I added an abstraction for GitPython to get committed files. I'm wondering if we can use one or the other implementations or combine them. Adding it as an abstraction allowed for easier unit testing.

Reply:
Good call out. I am planning on having a follow-up PR to optimize some things. Would you be OK with me addressing this in that follow-up PR? That way we can get the meat of the functionality in today while we're all still here, and then I can have a smaller PR that won't be so difficult to get reviews on from new Secureli folks.

Reply:
Captured here: https://github.com/slalombuild/secureli/issues/497

Reply:
Yeah, definitely!